src/cljs/app/events.cljs

```diff
@@ -0,0 +1,34 @@
+(ns app.events
+  "Handlers for global events across the app."
+    (:require [re-frame.core :as re-frame]
+              [app.subs :as subs]))
+
+(re-frame/reg-event-db
+ :initialize-db
+ (fn  [_ _]
+   subs/default-db))
+
+(re-frame/reg-event-db
+ :set-active-panel
+ (fn [db [_ active-panel]]
+   (assoc db :active-panel active-panel)))
+
+(re-frame/reg-event-db
+ :search/change-query
+ (fn [db [_ new-value]]
+   (assoc-in db [:search :query] new-value)))
+
+(re-frame/reg-event-db
+ :search/receive-results
+ (fn [db [_ search-results]]
+   (-> db
+       (assoc-in [:search :results] search-results)
+       (assoc :loading false))))
+
+(re-frame/reg-event-fx
+ :search/ragams!
+ (fn [{:keys [db]} [_ query]]
+   {:db (assoc db :loading true)
+    :http-get {:endpoint "search"
+               :query query
+               :type "ragam"}}))
```

Comment:
Would be good to pull out these functions, and give them names. Makes them more accessible, usable in the repl, and testable.

You can write these like so:
```clojure
(defn search-ragams [{:keys [db]} [_ query]]
   {:db (assoc db :loading true)
    :http-get {:endpoint "search"
               :query query
               :type "ragam"}})

(defn init []
 (re-frame/reg-event-fx :search/ragams! search-ragams)
 (re-frame/reg-event-fx :another-event another-event))
 ```

I'd suggest trying to keep all the code within functions wherever possible (like in `init` above). This gives better control over when they're run. "When this ns is required" is a rather crude, potentially confusing way of running code.